converter/ena/ena.py

```diff
@@ -0,0 +1,191 @@
+import logging
+import functools
+from api.ingest import IngestAPI
+from converter.ena.ena_experiment import EnaExperiment
+from converter.ena.ena_run import EnaRun
+from xsdata.formats.dataclass.serializers import XmlSerializer
+from xsdata.formats.dataclass.serializers.config import SerializerConfig
+
+
+class EnaArchive:
+    def __init__(self, uuid):
+        self.uuid = uuid
+        self.data = HcaData(uuid).get()
+        self.convert()
+
+    def convert(self):
+        config = SerializerConfig(pretty_print=True)
+        serializer = XmlSerializer(config=config)
+
+        experiment_set = EnaExperiment(self.data).create_set()
+        print(serializer.render(experiment_set))
+
+        run_set = EnaRun(self.data).create_set()
+        print(serializer.render(run_set))
+
+
+def handle_exception(f):
+    @functools.wraps(f)
+    def error(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except KeyError as e:
+            logging.error(f'HcaData KeyError: source entity does not contain key: {str(e)}')
+        except Exception as e:
+            logging.error(f'HcaData Exception: {str(e)}')
+
+    return error
+
+
+class HcaData(IngestAPI):
+    """
+    Assay:
+    [Biomaterial]  --> |   Process  |  -->  [File]
+                       |     |      |
+                       |     v      |
+                       | [Protocol] |
+    "submission": {
+        ...
+        "project": ...
+        "assays": [
+            {
+                ... # process
+                "sequencing_protocol": {...},
+                "library_preparation_protocol": {...},
+                "input_biomaterials": [...],
+                "derived_files": [...]
+            },
+            ...
+        ]
+    }
+    """
+    def __init__(self, uuid):
+        IngestAPI.__init__(self)
+        self.logger = logging.getLogger(__name__)
+        self.uuid = uuid
+
+    @handle_exception
+    def get(self):
+        url = f'{self.url}/submissionEnvelopes/search/findByUuidUuid?uuid={self.uuid}'
+        response = self.session.get(url, headers=self.headers)
+        self.submission = None
+
+        if response.ok:
+            self.submission = response.json()
+
+            # get project
+            projects = self.get_submission_entities('projects')
+            if projects:
+                if len(projects) == 1:
+                    self.submission["project"] = projects[0]
+                    self.logger.info(f'Project UUID {self.submission["project"]["uuid"]["uuid"]}.')
```

Comment:
I would change this to debug and also line 83.
I would also change line 85 to warning or error.